samples/vae-torch/trainer.py

```diff
@@ -0,0 +1,164 @@
+        """Create optimizers for the encoder and decoder models.
+
+        Returns:
+            OptimizersSetup: A dictionary containing the optimizer for the VAE.
+        """
+        params = itertools.chain(
+            self.encoder.model.parameters(),
+            self.decoder.model.parameters(),
+        )
+        self.optim = torch.optim.Adam(params, lr=self.lr)
+        return {"optimizer": self.optim}
+
+    @override
+    def train(self) -> None:
+        """Train the VAE model using the provided data.
+
+        This method retrieves the training data, sets up a DataLoader,
+        and performs the training loop, including forward passes, loss
+        computation, and backpropagation. It logs the training metrics
+        to TensorBoard.
+        """
+        data = self.data_user.get_data()["data"]
+        dataset = TensorDataset(torch.stack(list(data)))
+        dataloader = DataLoader(dataset, batch_size=self.batch_size, shuffle=True)
+
+        device = get_device(self.encoder.model)
+
+        batch_steps = 0
+
+        # Training loop for ordinary VAE
+        for epoch in range(self.max_epochs):
+            recon_sum = 0.0
+            kl_sum = 0.0
+            data_count = 0
+            for (batch,) in dataloader:
+                batch = batch.to(device)
+
+                # Forward pass
+                dist = self.encoder(batch)
+                z = dist.rsample()
+                recon = self.decoder(z)
+
+                # Reconstruction loss (L2 / MSE)
+                recon_loss = 0.5 * F.mse_loss(
+                    recon, batch, reduction="sum"
+                )  # Using half the MSE loss corresponds to
+                # assuming the decoder's output distribution has unit variance
+                # (i.e., N(recon, I)).
+
+                # KL divergence for Normal
+                mu = dist.loc
+                logvar = 2 * torch.log(dist.scale)
+                kl_loss = -0.5 * torch.sum(1 + logvar - mu.pow(2) - dist.scale.pow(2))
+
+                # Total loss
+                loss = recon_loss + kl_loss
+
+                # Backprop
+                self.optim.zero_grad()
+                loss.backward()
+                self.optim.step()
+
+                # accumulate metrics
+                batch_size = batch.size(0)
+                recon_sum += recon_loss.item()
+                kl_sum += kl_loss.item()
+                data_count += batch_size
+
+                # log per batch
+                self.writer.add_scalar(
+                    "Loss/Total/Batch", loss.item() / batch_size, batch_steps
+                )
+                self.writer.add_scalar(
+                    "Loss/Reconstruction/Batch",
+                    recon_loss.item() / batch_size,
+                    batch_steps,
+                )
+                self.writer.add_scalar(
+                    "Loss/KL/Batch", kl_loss.item() / batch_size, batch_steps
+                )
+                batch_steps += 1
```

Comment:
これは `self.global_steps`を定義して、それを利用してログしてください。

```suggestion
                self.writer.add_scalar(
                    "Loss/Total/Batch", loss.item() / batch_size, self.global_step
                )
                self.writer.add_scalar(
                    "Loss/Reconstruction/Batch",
                    recon_loss.item() / batch_size,
                    self.global_step,
                )
                self.writer.add_scalar(
                    "Loss/KL/Batch", kl_loss.item() / batch_size, self.global_step
                )
                self.global_step+= 1
```